### ISSUE TITLE
Maintain CustomerStorage counts

### ISSUE THREAD

**Issue by donaldgray** [dlcs/protagonist#353]
When an image is ingested into the Engine we verify that the size of the incoming asset won't exceed the allowance in the storage policy (via `ICustomerStorageRepository.VerifyStoragePolicyBySize`).

An `ImageStorage` record is written when saving both `I` and `T` family assets, the values set are:
* `ThumbnailSize` - `T` this is always 0. For `I` this is the total size of the thumbnail jpgs on disk, calculated before upload to S3.
* `Size` - For `T` this is the total size in S3 of all transcoded derivatives. For `I` it is the size of the generated derivative on disk, calculated before upload to S3.

However, the check to verify whether the file has been exceeded is done based on the size of the _origin_, rather than the derivatives that are actually stored which doesn't seem correct (e.g. a 2GB lossles audio file would be a lot smaller as an MP3).

The overall `CustomerStorage` table isn't updated either, this needs to be done as part of the ingest save (or periodically as part of the save). This would need done for both for the individual Space, and for the Customer. The alternative if doing it periodically would need to be weighted up with any potential issues resulting from users over exceeding their limit.

**Comment by donaldgray** [comment]
Do we need to account for assets that have directly been uploaded to the DLCS? Either by direct ingest, or via the Portal. These will accrue storage costs as they are not hosted at a separate origin.

**Comment by tomcrane** [comment]
Yes, we do - for some small users this would be their primary mechanism for getting stuff into the DLCS.

**Comment by tomcrane** [comment]
This will be refined further:

* Create image storage record
* Increment customerStorage record (not recalculate - job in #467 does that)
* Base on derivatives
* Ensure all families/types are counted correctly, not just images/jp2s
* Verify storage check is async

**Comment by donaldgray** [comment]
Some more background/details:

When an asset is stored, we update the `CustomerStorage` and `ImageStorage` records as part of the final save. See [/docs/storage.md](https://github.com/dlcs/protagonist/blob/main/docs/storage.md) for details of what values are stored.

This happens in the [EngineAssetRepository](https://github.com/dlcs/protagonist/blob/367d13ac5c74f4a6ca9ac13ba994206e9d882e88/src/protagonist/Engine/Data/EngineAssetRepository.cs#L54-L74), when..
* When the `IngestorExecutor` runs (either the full ingest is complete so immediately for `iiif-img` and `file` delivery-channels but later for `iiif-av` channel).
* When the `TimebasedIngestorCompletion` runs (after Transcode operation has completed for `iiif-av` channel).

The above means that we will have accurately stored the total bytes that once all processing is finished.

## Acceptance Criteria

* Update all ingestion processes to validate that customers haven't exceeded their storage allowance before ingesting any assets (ie call `VerifyFileSize` with a 0 size).
  * This will prevent an ingest process starting if already over allowance. No compute resources will be used prior to denying request.
* Update `Timebased` ingestion rules. Remove storage allowance check as it is using the size of the _origin_, which isn't accurate. We only store this for a short period - the size of transcode outputs is what we store.
